exercises/practice/bottle-song/bottle_song_test.go

```diff
@@ -0,0 +1,28 @@
+package bottlesong
+
+import (
+	"fmt"
+	"sort"
+	"testing"
+)
+
+func TestRecite(t *testing.T) {
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			actual := Recite(tc.input.startBottles, tc.input.takeDown)
+			if !equal(actual, tc.expected) {
+				t.Errorf("Recite(%d, %d) = %q, want: %q", tc.input.startBottles, tc.input.takeDown, actual, tc.expected)
+			}
+		})
+	}
+}
+
+func equal(a, b []string) bool {
+	if len(b) != len(a) {
+		return false
+	}
+
+	sort.Strings(a)
```

Comment:
In this exercise the order of the verses matter, so we shouldn't order the verses to see that they are equal. Checking if the slices are equal as is should be enough. Also, in general we want to consider a nil slice equal to an empty non-nil slice, so if both slices have length 0, consider them equal. We have something similar in other exercises.

Reply:
Good catch! I blindly copy-pasted this from another test 😅 
I've removed the sorts and added another if statement in